File: Reverse_List.cpp
void reverseList(Node** head) {
    Node* current = *head;
    Node* prev = nullptr;
    Node* next = nullptr;

    while (current != nullptr) {
        next = current->link;
        current->link = prev;
        prev = current;
        current = next;
    }

    *head = prev;
}

File: Reverse_List.java
public void reverseList(Node[] head) {
    Node current = head[0];
    Node prev = null;
    Node next = null;

    while (current != null) {
        next = current.link;
        current.link = prev;
        prev = current;
        current = next;
    }

    head[0] = prev;
}
